envoy/ssl/handshaker.h

```diff
@@ -180,6 +180,11 @@ class HandshakerFactory : public Config::TypedFactory {
   virtual SslCtxCb sslctxCb(HandshakerFactoryContext& handshaker_factory_context) const PURE;
 };
 
+// A handle tracking the certificate selection request. This can be used to supply additonal data
+// to attach to the TLS sockets, and to detect when a request is cancelled.
+class SelectionHandle {};
```

Comment:
I think this needs `virtual ~SelectionHandle() = default;` for cleanup to work properly